File: java/socketd/src/main/java/org/noear/socketd/transport/core/buffer/BufferReader.java
package org.noear.socketd.transport.core.buffer;

/**
 * 缓冲读
 *
 * @author noear
 * @since 2.0
 */
public interface BufferReader {

    /**
     * 获取 byte
     */
    byte getByte();

    /**
     * 获取一组 byte
     */
    void getBytes(byte[] dst, int offset, int length);

    /**
     * 获取 int
     */
    int getInt();

    /**
     * 剩余长度
     */
    int remaining();

    /**
     * 当前位置
     */
    int position();
}

File: javascript/ts/org/noear/socketd/transport/core/buffer/BufferReader.ts
/**
 * 缓冲读
 *
 * @author noear
 * @since 2.0
 * @class
 */
export interface BufferReader {

    /**
     * 获取 byte
     */
    get(): number

    /**
     * 获取一组 byte
     */
    getBytes(dst: ArrayBuffer, offset: number, length: number);

    /**
     * 获取 int
     * @return {number}
     */
    getInt(): number;

    /**
     * 剩余长度
     * @return {number}
     */
    remaining(): number;

    /**
     * 当前位置
     * @return {number}
     */
    position(): number;
}